<!-- Gaussian Process-based Traversability Analysis for Terrain Mapless Navigation (ICRA 2024) -->
<!-- Abe Leininger, Mahmoud Ali, Hassan Jardali, Lantao Liu -->

<launch>
    <!-- Gaussian Process Mapping Module -->
    <node name="gp_mapping_module" pkg="gp_navigation" type="gp_mapping_module.py" output="screen">
        <rosparam command="load" file="$(find gp_navigation)/config/gp_module/gp_module.yaml"/>
    </node>

    <!-- Path Planning Module -->
    <node name="path_planning_module_client" pkg="gp_navigation" type="path_planning_module_client.py" output="screen">
        <rosparam command="load" file="$(find gp_navigation)/config/gp_module/gp_module.yaml"/>
        <rosparam command="load" file="$(find gp_navigation)/config/path_planning/jackal_env_a.yaml"/>
    </node>
    <node name="path_planning_module_server" pkg="gp_navigation" type="path_planning_module_server.py" output="screen"/>

    <!-- Differential Drive Controller -->
    <node name="diff_drive_go_to_goal" pkg="diff_drive" type="diff_drive_go_to_goal" output="screen">
        <rosparam command="load" file="$(find gp_navigation)/config/diff_drive/diff_drive.yaml"/>
        <param name="~forwardMovementOnly" value="false"/>
        <remap from="odom" to="ground_truth/state"/>
    </node>

    <!-- Transformations -->
    <node name="transform_node" pkg="gp_navigation" type="transform_node.py"/>

    <!-- Visualization -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find gp_navigation)/rviz_config/gp.rviz" required="true"/>
</launch>